File: pizzaria-front/pizzaria-front/src/app/models/pizza/pizza.ts
import { Sabor } from "../sabor/sabor";

export class Pizza {
    id!: number;
    ativo!: boolean;
    sabor!: Sabor[];
    tamanho!: string;
    qtdeSabor!: number;
    valorPizza!: number;
    registro!: Date;
    atualizar!: Date;

    constructor() {
        this.sabor = [];
    }
}

File: pizzaria/src/main/java/br/com/uniamerica/pizzaria/entity/Pizza.java
package br.com.uniamerica.pizzaria.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tb_pizza", schema = "public")
public class Pizza extends AbsctractEntity{
    @Getter
    @Setter
    @ManyToMany
    @JoinTable(name = "cl_pizza.sabor",
            joinColumns =  @JoinColumn(
                    name = "pizza.id"
            ),
            inverseJoinColumns = @JoinColumn(
                    name = "sabor.id"
            )
    )
    private List<Sabor> sabor = new ArrayList<>();

    @Getter @Setter
    @Enumerated(EnumType.STRING)
    @Column(name = "cl_tamanho")
    private Tamanho tamanho;

    @Getter @Setter
    @Column(name = "cl_qtde_sabor")
    private int qtdeSabor;

    @Getter @Setter
    @Column(name = "cl_valor_pizza")
    private BigDecimal valorPizza;

}
